server/lib/tuist/clickhouse_repo.ex

```diff
@@ -7,4 +7,17 @@ defmodule Tuist.ClickHouseRepo do
     otp_app: :tuist,
     adapter: Ecto.Adapters.ClickHouse,
     read_only: true
+
+  @task_timeout 60_000
+  def parallel_tasks(queries, opts \\ []) do
+    max_concurrency = Keyword.get(opts, :max_concurrency, 3)
+
+    queries
+    |> Task.async_stream(fn fun -> fun.() end,
+      max_concurrency: max_concurrency,
+      timeout: @task_timeout
+    )
+    |> Enum.to_list()
+    |> Enum.map(fn {:ok, result} -> result end)
```

Comment:
Should we iterate a bit on this to return:

```exs
{:ok, resulst}
{:error, errors}
```

Reply:
> I don't think that's particularly valuable as we don't have a way to show the user "This specific widget failed to load its data." - this will show a 500 with a match error, and I think that's okay.

I agree. Let's keep it like that.